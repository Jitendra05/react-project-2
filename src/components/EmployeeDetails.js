import React from "react";

import DataTable from "react-data-table-component";

const employees = [
  { id: 1, name: "jitendra", doj: "1981", role: "dev", salary: 1000 },
  { id: 2, name: "vijay", doj: "1982", role: "tester", salary: 2000 },
  { id: 3, name: "ajay", doj: "1983", role: "dev", salary: 3000 },
  { id: 4, name: "ram", doj: "1982", role: "tester", salary: 2000 },
  { id: 5, name: "jitendra", doj: "1981", role: "dev", salary: 1000 },
  { id: 6, name: "vijay", doj: "1982", role: "tester", salary: 2000 },
  { id: 7, name: "ajay", doj: "1983", role: "dev", salary: 3000 },
  { id: 8, name: "ram", doj: "1982", role: "tester", salary: 2000 },
  { id: 9, name: "jitendra", doj: "1981", role: "dev", salary: 1000 },
  { id: 10, name: "vijay", doj: "1982", role: "tester", salary: 2000 },
  { id: 11, name: "ajay", doj: "1983", role: "dev", salary: 3000 },
  { id: 12, name: "ram", doj: "1982", role: "tester", salary: 2000 },
  { id: 13, name: "jitendra", doj: "1981", role: "dev", salary: 1000 },
  { id: 14, name: "vijay", doj: "1982", role: "tester", salary: 2000 },
  { id: 15, name: "ajay", doj: "1983", role: "dev", salary: 3000 },
  { id: 16, name: "ram", doj: "1982", role: "tester", salary: 2000 }
];
const columns = [
  {
    name: "Id",
    selector: "id",
    sortable: true
  },
  {
    name: "Name",
    selector: "name",
    sortable: true,
    right: true
  },
  {
    name: "Designation",
    selector: "role",
    sortable: true,
    right: true
  },
  {
    name: "Salary",
    selector: "salary",
    sortable: true,
    right: true
  },
  {
    name: "Joining Date",
    selector: "doj",
    sortable: true,
    right: true
  }
];

class EmployeeDetails extends React.Component {
  handleChange = state => {
    // You can use setState or dispatch with something like Redux so we can use the retrieved data
    console.log("Selected Rows: ", state.selectedRows);
  };

  render() {
    return (
      <DataTable
        title="Employee Details"
        columns={columns}
        data={employees}
        selectableRows
        onRowSelected={this.handleChange}
        pagination
      />
    );
  }
}

export default EmployeeDetails;
