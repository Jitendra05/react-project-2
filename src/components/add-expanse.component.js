import React from 'react';
import ExpanseForm from './expanse-form.component';
import {startAddExpanse} from '../actions/expanse-action';
import {connect} from 'react-redux';

export class  AddExpanseComponent extends React.Component{
    onSubmit = (expanse) => {
        this.props.startAddExpanse(expanse);
        this.props.history.push('/');
    }
    render() {
        return  (
            <div>
                <div className="page-header">
                    <div className="container">
                        <h2 className="page-header__title">
                            Add Expense
                        </h2>
                    </div>
                </div>
                <div className="container">
                    <ExpanseForm onSubmit={this.onSubmit}/>
                </div>
            </div>
        );
    }
}


const mapDispatchToProps = (dispatch) => ({
    startAddExpanse: (expanse) => dispatch(startAddExpanse(expanse))
});

export default connect(undefined,mapDispatchToProps)(AddExpanseComponent);