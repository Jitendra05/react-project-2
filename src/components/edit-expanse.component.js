import React from 'react';
import {connect} from 'react-redux';
import ExpanseForm from './expanse-form.component';
import {startEditExpanse} from '../actions/expanse-action';
import WarningModal from './modals/warning-popup.modal';
import SuccessModal from './modals/success-popup.modal';

export class  EditExpanseComponent extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isSuccess:false,
            isWarning:false,
            id:undefined,
            updatedExpanse:undefined
        }
    }

    closePopup = () => {
        this.setState(() => ({isSuccess:false,isWarning:false}));
    }
    okPopup = () => {
        this.setState(()=> ({isSuccess:true,isWarning:false}));
        setTimeout(() => {
            this.props.startEditExpanse(this.state.id,this.state.updatedExpanse);
            this.props.history.push('/');
        }, 1000);
    }
    componentWillUnmount() {
        this.setState(()=> ({isSuccess:false, isWarning:false}));
    }
    onSubmit = (updatedExpanse) => {
        this.setState(()=> ({
             isSuccess:false,
             isWarning:true,
             id:this.props.expanse.id,
             updatedExpanse
         }));
    }
    render() {
        return (
            <div className="container">
               <ExpanseForm expanse={this.props.expanse} onSubmit={this.onSubmit} 
               />
                <WarningModal 
                    id="warningModal"
                    isVisible={this.state.isWarning}
                    title='Warning' 
                    message='Are you sure? You want to update the expanse.' 
                    closePopup={this.closePopup}
                    okPopup={this.okPopup}
                />
                <SuccessModal
                    id="SuccessModal" 
                    isVisible={this.state.isSuccess}
                    title='Success' 
                    message='Expanse updated sucessfuly.'
                />
            </div>
        );
    }
}

const mapStateToProps = (state, props) => {
    return {
        expanse: state.expanses.find((expanse)=> expanse.id===props.match.params.id)
    };
};


const mapDispatchToProps = (dispatch) => {
    return {
        startEditExpanse: (id,updatedExpanse) => dispatch(startEditExpanse(id,updatedExpanse))
    };
};

export default connect(mapStateToProps,mapDispatchToProps)(EditExpanseComponent);