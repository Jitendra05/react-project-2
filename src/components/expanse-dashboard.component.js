import React from 'react';
import ExpanseList from './expanse-list.component';
import ExpanseListFilter from './expanse-list-filter.component';
import ExpanseSummary from '../components/expanse-summary.component';

const ExpanseDashboardComponent = () => (
    <div>
        <ExpanseSummary />
        <ExpanseListFilter />
        <div className="container">
         <ExpanseList />
        </div>
    </div>
);

export default ExpanseDashboardComponent;