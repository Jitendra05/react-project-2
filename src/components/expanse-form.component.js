import React from 'react';
import moment from 'moment';
import {SingleDatePicker} from 'react-dates';

export default class ExpanseForm extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            description: props.expanse ? props.expanse.description:'',
            amount:props.expanse ? props.expanse.amount.toString():'',
            note:props.expanse ? props.expanse.note:'',
            createdAt:props.expanse ? moment(props.expanse.createdAt):moment(),
            focused:false,
            error: ''
        };
    }


    onDescriptionChange = (e) => {
       const description = e.target.value;
       this.setState(()=>({description}));
    }

    onAmountChange = (e) => {
        const amount = e.target.value;
        if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(()=>({amount}));
        }
     }

     onNoteChange = (e) => {
        const note = e.target.value;
        this.setState(()=>({note}));
     }

     onSubmit = (e) => {
        e.preventDefault();
        if(!this.state.description || !this.state.amount) {
            this.setState(()=>({error:'Please provide the descrption and amount.'}));
        } else {
            this.setState(()=>({error:''}));
            const expanse = {
                description: this.state.description,
                amount: parseFloat(this.state.amount),
                createdAt: this.state.createdAt.valueOf(),
                note: this.state.note
            };
            this.props.onSubmit(expanse);

        }
     }

    render() {
        return (
                <form className="from-view" onSubmit={this.onSubmit}>
                {this.state.error && <h4 className="form-error">{this.state.error}</h4>}
                <div className="text-label ">
                        <label>Description:</label>
                    </div>
                    <input 
                        type="text" 
                        autoFocus
                        className="text-input"
                        placeholder="Description" 
                        value={this.state.description}
                        onChange={this.onDescriptionChange}
                    />
                    
                    <div className="text-label ">
                        <label>Amount:</label>
                    </div>
                    <input 
                        type="number" 
                        className="text-input"
                        placeholder="Amount"
                        value={this.state.amount}
                        onChange={this.onAmountChange}
                    />
                   <div className="text-label ">
                        <label>Created At:</label>
                    </div>
                    <SingleDatePicker
                            date={this.state.createdAt} // momentPropTypes.momentObj or null
                            onDateChange={createdAt =>{
                                if(createdAt){
                                    this.setState({ createdAt })
                                }
                            }} // PropTypes.func.isRequired
                            focused={this.state.focused} // PropTypes.bool
                            onFocusChange={({ focused }) => this.setState({ focused })} // PropTypes.func.isRequired
                            id="someid" // PropTypes.string.isRequired,
                            numberOfMonths={1}
                            isOutsideRange={() => false}
                    />
                    <div className="text-label ">
                        <label>Note:</label>
                    </div>
                    <textarea 
                        placeholder="write a note on your expanse(optional)"
                        className="text-area"
                        value={this.state.note}
                        onChange={this.onNoteChange}
                    />
                    <div>
                        <button className="large-button">Save Expense</button>
                    </div>
                </form>
        );
    }
}