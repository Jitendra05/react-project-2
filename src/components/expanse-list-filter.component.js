import React from 'react';
import {connect} from 'react-redux';
import {setFilterText, sortByAmount, sortByDate, setStartDate, setEndDate} from '../actions/filter-action';
import {DateRangePicker} from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';
import 'react-dates/initialize';

class ExpanseListFilterComponent extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            startDate:props.filters.startDate,
            endDate:props.filters.endDate,
            focusedInput:null
        };
    }
    render() {
        return (
            <div>
                <input type="text" placeholder="search on descption" value={this.props.filters.text} onChange={(e) => {
                    this.props.dispatch(setFilterText(e.target.value));
                }}/>
                <select value={this.props.filters.sortBy} onChange={(e) => {
                    if(e.target.value ==='date') {
                        this.props.dispatch(sortByDate());
                    } else if(e.target.value ==='amount') {
                        this.props.dispatch(sortByAmount());
                    }
                }}>
                    <option value='date'>Date</option>
                    <option value='amount'>Amount</option>
                </select>
                <DateRangePicker
                    startDate={this.state.startDate} // momentPropTypes.momentObj or null,
                    startDateId="your_unique_start_date_id" // PropTypes.string.isRequired,
                    endDate={this.state.endDate} // momentPropTypes.momentObj or null,
                    endDateId="your_unique_end_date_id" // PropTypes.string.isRequired,
                    onDatesChange={({ startDate, endDate }) => {
                        this.setState(()=>({startDate,endDate}));
                        this.props.dispatch(setStartDate(startDate));
                        this.props.dispatch(setEndDate(endDate));
                        } // PropTypes.func.isRequired,
                    }
                    focusedInput={this.state.focusedInput} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
                    onFocusChange={focusedInput => this.setState({ focusedInput })} // PropTypes.func.isRequired,
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                    showClearDates={true}
                    
                />  
            </div>
        );
    }
}  

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    } ;
}

export default connect(mapStateToProps)(ExpanseListFilterComponent);