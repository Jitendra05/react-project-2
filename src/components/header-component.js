import React from 'react';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import {startLogout} from '../actions/auth-action';

export const HeaderPage = ({startLogout}) => (
      <div className="header">
        <div className="container">
        <div className="header__content">
            <Link className="header__title" to="/dashboard" >
              <h1>Expensify</h1>
            </Link>
            <button 
            onClick={startLogout}
              className="button button--link" 
            >
            Logout
            </button>
        </div>
           
        </div>
      </div>
);

const mapDispatchToProps =  (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined,mapDispatchToProps)(HeaderPage);
