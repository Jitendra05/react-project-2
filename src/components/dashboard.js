import React from 'react';
import Header from './header-component';;
import EmployeeDetails from './EmployeeDetails';
import SampleModal from "./modals/SampleModal";
import OptionModal from './modals/OptionModal';
import '../styles/style.scss'
import Footer from './footer.component';


export  class  Welcome extends React.Component {
  render() {
    return (
    <div>
      <h3>Hello {this.props.name}</h3>
    </div>
    );
  }
};

export default class DashboardApp extends React.Component {
  state = {
    time : new Date().toLocaleTimeString()
  }
  tick = ()  => {
   this.setState(()=> ({time: new Date().toLocaleTimeString()}));
  }



  render() {
    setInterval(this.tick,1000);
    const subtitle = 'Check your expanses...';
    return (
      <div>
        <Header subtitle={subtitle} />
        <div className="container">
        <h2>Dashboard</h2>
        <Welcome name={'jitu'}/>

        <h3> Time is : {this.state.time}</h3>
          {/* <EmployeeDetails />
          <SampleModal /> */}
        </div>
        <Footer />
        <OptionModal />                
      </div>
    );
  }
}
