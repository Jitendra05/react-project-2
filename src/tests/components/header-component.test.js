import React from 'react';
import {HeaderPage} from '../../components/header-component';
import {shallow} from 'enzyme';

test('should render HeaderPage component correctly',()=>{
    const wrapper = shallow(<HeaderPage />);
    expect(wrapper).toMatchSnapshot();
    // expect(wrapper.find('h1').text()).toBe('Expansify');
});
