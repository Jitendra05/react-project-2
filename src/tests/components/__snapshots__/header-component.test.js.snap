// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HeaderPage component correctly 1`] = `
<div
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        to="/dashboard"
      >
        <h1>
          Expensify
        </h1>
      </Link>
      <button
        className="button button--link"
      >
        Logout
      </button>
    </div>
  </div>
</div>
`;
